File: blockchain/02-OutputsInputsTransactions/src/main/java/no/kalli/CoinbaseTx.java
package no.kalli;

public class CoinbaseTx {

    //Simplified compared to Bitcoin (nothing significant missing)
    private String coinbase; // "The Times 03/Jan/2009 Chancellor
    //  on brink of second bailout for banks"
    private Output output;
    private String txHash;

    public CoinbaseTx(String coinbase, int value, String address) {
        //TODO
        //Remember to calculate txHash
    }

    @Override
    public String toString() {
        //TODO For screen output
        return "";
    }

    //TODO Getters?

}

File: blockchain/02-OutputsInputsTransactions/src/main/java/no/kalli/CoinbaseTx.kt
package no.kalli

data class CoinbaseTx(var coinbase: String, private var value: Long, private var address: String) {

    private val txHash: String
        get() = HashUtil.base64Encode(HashUtil.sha256Hash(coinbase))

    val output: Output = Output(value, address)
}